README.md

```diff
@@ -162,39 +230,81 @@ interface IDXOptions {
 
 **Returns** `Promise<Definition>`
 
-#### .getEntryId
+#### .getEntry
 
 **Arguments**
 
 1. `definitionId: DocID`
-1. `did?: string`
+1. `did?: string = this.id`
 
-**Returns** `Promise<DocID | null>`
+**Returns** `Promise<Entry | null>`
 
-#### .getEntry
+#### .getEntryReference
 
 **Arguments**
 
 1. `definitionId: DocID`
-1. `did?: string`
+1. `did?: string = this.id`
 
 **Returns** `Promise<unknown | null>`
 
+#### .getEntryTags
+
+**Arguments**
+
+1. `definitionId: DocID`
+1. `did?: string = this.id`
+
+**Returns** `Promise<Array<string>>`
+
 #### .setEntry
 
 **Arguments**
 
+1. `definitionId: DocID`
+1. `entry: Entry`
+
+**Returns** `Promise<void>`
+
+#### .setEntryReference
+
+Sets the content of the entry reference.
+
+> The provided tags will only be set if the entry is getting created, if it already exists tags will not be changed
+
+**Arguments**
+
 1. `definitionId: DocID`
 1. `content: unknown`
+1. `tags?: Array<string> = []`
 
 **Returns** `Promise<DocID>` the `DocID` of the created content document
 
+#### .addEntryTag
+
+**Arguments**
+
+1. `definitionid: DocID`
+1. `tag: string`
+
+**Returns** `Promise<Array<string>>` the updated set of tags
+
+#### .removeEntryTag
+
+**Arguments**
+
+1. `definitionId: DocID`
+1. `tag: string`
+
+**Returns** `Promise<Array<string>>` the updated set of tags
```

Comment:
These seem redundant?

Reply:
Yes could be simply returning `void`, but I think it's convenient not to have to apply the same changes in the apps logic, as they're performed in the lib anyways.

Reply:
I tend to be more on the side of having more methods to do the same thing only confuses developers.